### PR TITLE
Sem 07 tasks

### PR DESCRIPTION
**Pull request by GeorgiTerziev02**
I can optimize the tasks not to use strLen if needed.

### CODE REVIEW

File: IS/Sem.07/Solutions.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    while (str1[i] == str2[i]) {
+        if (str1[i] == '\0'/* && str2[i] == '\0'*/) {
+            return 0;
+        }
+
+        i++;
+    }
+
+    return str1[i] - str2[i] > 0 ? 1 : -1;
```
Comment (Justsvetoslavov) [inline IS/Sem.07/Solutions.cpp:24]: Ternary is not necessary.
Just return str1[i] - str2[i]

File: IS/Sem.07/Solutions.cpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+void myStrCat(const char* str1, const char* str2, char* result) {
```
Comment (Justsvetoslavov) [inline IS/Sem.07/Solutions.cpp:27]: We can use only dest and source.

Suggestion:

```
void my_strcat(char* dest, const char* source)
{
	size_t destLength = my_strlen(dest);
	size_t sourceLength = my_strlen(source);

	for (size_t i = 0; i < sourceLength; i++)
	{
		dest[destLength + i] = source[i];
	}
	dest[destLength + sourceLength] = '\0';
}
```

File: IS/Sem.07/Solutions.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    for (size_t i = 0; i < str2Length; i++) {
+        result[str1Length + i] = str2[i];
+    }
+
+    result[str1Length + str2Length] = '\0';
+}
+
+
+void myStrCopy(char* destination, const char* source) {
```
Comment (Justsvetoslavov) [inline IS/Sem.07/Solutions.cpp:43]: rename to dest (not necessary, as it is a preference)

**Review by Justsvetoslavov — COMMENTED** [review]
Should we consider splitting into multiple .cpps ,as they are different tasks?

File: IS/Sem.07/Solutions.cpp (hunk 1 above)
Comment (GeorgiTerziev02) [reply IS/Sem.07/Solutions.cpp:24]: The students wanted it with -1 and 1 :D

**Review by Justsvetoslavov — APPROVED** [review]
👏

**Pull request merged**